### DISCUSSION TITLE
YimMenu Lua Only Menu

### DISCUSSION THREAD

**Discussion by Deadlineem** [YimMenu/YimMenu#3206] [Q&A]
How hard would it be to make a version of YimMenu that only relies on LUA scripts? 

Not in the sense that it runs entirely off of it, of course YimMenu would be the base and the lua aspect would be the main functionality for creating new tabs/options.  Im considering gutting Yim completely (aside from the necessary bits of it like hooking and pointers that were mentioned) and putting just the lua functionality into it / improving on it so people can compile it and then simply make their own "menu" out of it through lua scripting alone.  I figured id ask this here since it is not an issue, just a simple question as to how difficult it would be to do,

**Comment by Rxann** [comment]
I would say it would be most likely extremely hard.

**Reply by Deadlineem** [reply]
> I would say it would be most likely extremely hard.

I was thinking that but also, since everything is already there the yim devs could probably just gut the menu tabs (aside from the settings tab entirely, thats needed) and all the related stuff to those tabs (Self, Weapons, Vehicle, World, etc.) and keep the lua functionality and bindings for it.

I personally have never came across a standalone lua based menu so it would be neat imo.

**Reply by Rxann** [reply]
Yeah, most of the issues would be hooking and pointers. That would be where you would have the most issues.

**Reply by Deadlineem** [reply]
> Yeah, most of the issues would be hooking and pointers. That would be where you would have the most issues.

I made my post a bit more specific, gutting out the tabs and options and just focusing on lua, if i had info on how i could do this, and maybe some others interested in the project, it could be a thing, the lua aspect is amazing already it could help with the functionality of the main branches lua along side it.
